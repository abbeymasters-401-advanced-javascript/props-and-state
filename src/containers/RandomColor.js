import React, { Component } from 'react';

export default class RandomColor extends Component {

  state = { backgroundColor: 'black', backgroundImg: '' };
  
  pickColor = (colors) => {
    let newBackground = colors[Math.floor(Math.random() * colors.length)];
    if(this.state.backgroundColor === newBackground) this.setState({ backgroundImg: 'url(https://i.dailymail.co.uk/i/pix/2016/03/18/15/324D202500000578-3498922-image-a-33_1458315465874.jpg)' });
    else this.setState({ backgroundColor: newBackground, backgroundImg: '' });
  };

  componentDidMount() {
    const colors = ['red', 'blue', 'green', 'purple', 'yellow', 'orange', 'pink', 'black', 'aqua'];
    window.setInterval(() => this.pickColor(colors), 1000);
  }

  render() {
    const bodyStyle = {
      backgroundColor: this.state.backgroundColor,
      backgroundImage: this.state.backgroundImg,
      width: '1450px',
      height: '1000px'
    };

    return (
      <div style={bodyStyle}></div>
    );
  }
}
