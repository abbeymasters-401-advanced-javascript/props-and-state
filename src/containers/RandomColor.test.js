import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('Random Color component', () => {
  it('renders randomcolor', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
