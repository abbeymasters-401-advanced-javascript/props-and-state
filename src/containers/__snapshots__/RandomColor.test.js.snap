// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Color component renders randomcolor 1`] = `
<div
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "",
      "height": "1000px",
      "width": "1450px",
    }
  }
/>
`;
